### PR TITLE
Randomize bernoulli and dropout ops.

### PR DESCRIPTION
**Pull request by dlibenzi**
We used to embed the RNG seeds within the graphs, and in order to avoid recompilations, we'd reset the master seed at every step.
This makes compilations stable, but things are not working as one would expect.
Every step will generate the same random numbers at a given position in the graph, which means things like Dropout will not work as intended.
This PR uses an XLA parameter (device data) as master seed, which gets pushed forward at every step, allowing stable compilations and progressive number generation.

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -1281,7 +1308,7 @@ void XLATensor::SyncLiveTensorsGraph(const Device* device,
 
 void XLATensor::MarkStep(const Device* device) {
   XLA_COUNTER("MarkStep", 1);
-  DeviceContextArena::Get()->ClearProfileData(device);
+  DeviceContextArena::Get()->StepRngSeed(device);
```
Comment (JackCaoG) [inline torch_xla/csrc/tensor.cpp:1311]: wondering why is this needed. Even without this, everytime someone call `GetRngSeed`, they will get a new seed right?

File: torch_xla/core/xla_model.py [hunk 2]
```diff
@@ -395,9 +395,9 @@ def all_to_all(value,
     split_count (int): The split count.
     groups (list, optional): A list of list, representing the replica groups for
       the `all_reduce()` operation. Example: `[[0, 1, 2, 3], [4, 5, 6, 7]]`
-      defines two groups, one with the `[0, 1, 2, 3]` replicas and one with the
-      `[4, 5, 6, 7]` replicas. If `None` there will be only one group with all
-      the replicas in it.
+        defines two groups, one with the `[0, 1, 2, 3]` replicas and one with
+        the `[4, 5, 6, 7]` replicas. If `None` there will be only one group with
```
[the remark this answers is not in the document]
Comment (dlibenzi) [reply torch_xla/core/xla_model.py:399]: I noticed that but we need to leave the formatting alone and not manually fixup, otherwise there would be no point of having automatic formatters.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp:1311]: This has the same reason of `mark_step()` for tensors.
If we do not "trim" the IR chain, the seed IR graph keeps growing from the master seed (device data) down to its current head.
